### ISSUE TITLE
Define training file format

### ISSUE THREAD

**Issue by MartinThoma** [MartinThoma/nntoolkit#3]
Training files for classification formats have to have

* The label
* The features (at least one feature, but arbitrary many)

Open questions:

* How does training data look like for regression tasks and how is it stored best?
* What are relevant applications of neural networks besides classification and regression?

Relevant properties for the choice of the format are:

* The format should be open and language-independent
* The format should be well-known if possible
* The format should be small in size, as some tasks might have A LOT of training data
* Reading is more important than writing, because the training format is only used while training. At training time, many different options might get considered and the same file might be used very often.
